client-common/src/tendermint/types/block_results.rs

```diff
@@ -20,7 +20,10 @@ pub trait BlockResults {
 
     /// Checks if a StakedStateAddress is included in devlier_tx account event.
     /// Returns true when the address presents
-    fn contains_account(&self, target_account: &StakedStateAddress) -> Result<bool>;
+    fn contains_account(
+        &self,
+        contain_account_callback: Box<dyn Fn(StakedStateAddress) -> bool>,
```

Comment:
Can you add some documentation about what this callback function should do and also why this callback pattern is necessary?

Reply:
that's because, BlockResults cannot have ownership for DB, also it's complicated to give trait ownership to it.
so decoupling between BlockResults and filter logic in DB

Reply:
Ok